logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -47,6 +47,10 @@ def initialize(params={})
 
   def register
   end
+  
+  def threadsafe?
+    false
```

Comment:
I don't think this currently exists as an API. In https://github.com/elastic/logstash/pull/4391/files#diff-3d93d6656c81049b6de7a09dbe141a46R30 I added a new method `Outputs/Base.declare_threadsafe!` that would declare this at the class level.